website/pages/docs/concepts/password-policies.mdx

```diff
@@ -0,0 +1,290 @@
+letter, at least one number, and at least one symbol from the set `!@#$%^&*`:
+
+```hcl
+length = 20
+rule "charset" {
+  charset = "abcdefghijklmnopqrstuvwxyz"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "0123456789"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "!@#$%^&*"
+  min-chars = 1
+}
+```
+
+At least one charset must be specified for a policy to be valid. In order to generate a password, a charset
+must be available to select characters from and these policies do not have a default charset specified.
+The following policy is **NOT** valid and will be rejected:
+
+```hcl
+length = 20
+```
+
+## Performance Characteristics
+Characterizing password generation performance with this model is heavily dependent on the policy
+configuration. In short, the more restrictive the policy, the longer it will take to generate a password.
+This generalization isn't always true, but is a general guideline.
+
+Here are some examples policy configurations with their performance characteristics below:
+
+<details>
+<summary>No Restrictions (94 chars)</summary>
+
+```hcl
+rule "charset" {
+  charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~"
+}
+```
+</details>
+
+<details>
+<summary>One each: Alphabetic, Numeric, and '-' (63 chars)</summary>
+
+```hcl
+rule "charset" {
+  charset = "abcdefghijklmnopqrstuvwxyz"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "0123456789"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "-"
+  min-chars = 1
+}
+```
+</details>
+
+<details>
+<summary>One each: Alphabetic, Numeric & All ASCII Symbols (94 chars)</summary>
+
+```hcl
+rule "charset" {
+  charset = "abcdefghijklmnopqrstuvwxyz"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "0123456789"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~"
+  min-chars = 1
+}
+```
+</details>
+
+<details>
+<summary>One each: Alphabetic & a subset of unicode characters (256 chars)</summary>
+
+```hcl
+rule "charset" {
+  charset = "abcdefghijklmnopqrstuvwxyz"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+  min-chars = 1
+}
+rule "charset" {
+  charset = "ĩĪīĬĭĮįİıĲĳĴĵĶķĸĹĺĻļĽľĿŀŁłŃńŅņŇňŉŊŋŌōŎŏŐőŒ"
+  min-chars = 1
+}
+
+# Filling in the rest of the charset to 256 characters
+rule "charset" {
+  charset = "0123456789"
+}
+rule "charset" {
+  charset = " !\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~"
+}
+rule "charset" {
+  charset = "ĀāĂăĄąĆćĈĉĊċČčĎďĐđĒēĔĕĖėĘęĚěĜĝĞğĠġĢģĤĥĦħĨœŔŕŖŗŘřŚśŜŝŞşŠšŢţŤťŦŧŨũŪūŬŭŮůŰűŲųŴŵŶŷŸŹźŻżŽžſ℀℁ℂ℃℄℅℆ℇ℈℉ℊℋℌℍℎℏℐℑℒℓ℔ℕ№℗℘ℙℚℛℜℝ℞℟℠"
+}
+```
+</details>
+
+[![Password Policy Performance](/img/vault-password-policy-performance.svg)](/img/vault-password-policy-performance.svg)
```

Comment:
I didn't want to put specific numbers here to prevent questions around "why didn't it perform as described in the docs?" when it's highly dependent on what machine you're running it on, the load on Vault, etc. I mainly wanted to show the relationship between some common configurations. I can put a label on the Y axis if that helps clarify it.

Reply:
You could potentially handle it via having a relative averages of the values, picking the mean as 1.0 and adjusting the range for std deviations or some such... but... I'm no statistician.

Reply:
Yep, makes sense to not put down specific numbers. The label on the Y axis looks good.